Support multipart/mixed in Rest Client Reactive

### Description

I would be happy If I could send multipart/mixed requests using Rest Client Reactive rather than using Apache HttpClient. Therefore I request this feature!

---

Discussions: https://github.com/quarkusio/quarkus/discussions/28021
Sample Project (I researched the current status): https://github.com/Ballsigno/quarkus-multipart-mixed-test

### Implementation ideas

_No response_

@geoand 
I'm sorry, my memory is a bit fuzzy. I just wanted to use "multipart/mixed" like "multipart/form-data".  
Those methods are traces of looking for a workaround. [This comment](https://github.com/Ballsigno/quarkus-multipart-mixed-test/blob/main/rest-client-reactive/src/main/java/org/test/service/RestService.java#L29) is the trigger.

I think the ideal solution would be to use "mixed" like "form-data".

```java
@POST
@Consumes("multipart/mixed") 
@Produces(MediaType.TEXT_PLAIN)
String postMultipartMixed(@MultipartForm MultipartBody data);
```

Currently, the following errors occur.

```txt
Caused by: java.lang.RuntimeException: '@MultipartForm' can only be used on methods annotated with '@Consumes(MediaType.MULTIPART_FORM_DATA)'. Offending method is 'org.test.service.RestService#java.lang.String postMultipartMixed(org.test.model.MultipartBody data)'
```

--- a/extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/multipart/MultiByteFileTest.java
+++ b/extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/multipart/MultiByteFileTest.java
@@ -15,6 +15,7 @@ import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
 import jakarta.ws.rs.Consumes;
 import jakarta.ws.rs.FormParam;
+import jakarta.ws.rs.HeaderParam;
 import jakarta.ws.rs.POST;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.core.MediaType;
@@ -139,6 +140,17 @@ public class MultiByteFileTest {
         assertThat(result).isEqualTo("myFile");
     }
 
+    @Test
+    public void shouldPropertySetContentType() {
+        Client client = RestClientBuilder.newBuilder().baseUri(baseUri).build(Client.class);
+
+        ClientForm form = new ClientForm();
+        form.file = Multi.createBy().repeating().supplier(() -> (byte) 4).atMost(100);
+
+        assertThat(client.postMultipartToType(form)).startsWith("multipart/form-data");
+        assertThat(client.postMultipartMixedToType(form)).startsWith("multipart/mixed");
+    }
+
     private void delayedEmit(AtomicLong i, MultiEmitter<? super Byte> em) {
         vertx.setTimer(100, id -> {
             long index = i.getAndIncrement();
@@ -188,6 +200,12 @@ public class MultiByteFileTest {
         public String uploadEmpty(@MultipartForm FormWithTwoFiles form) {
             return verifyFile(form.file1, 100, position -> (byte) 4)
                     + verifyFile(form.file2, 100, position -> (byte) 7);
+        }
+
+        @Path("type")
+        @POST
+        public String contentType(@HeaderParam("Content-Type") String contentType) {
+            return contentType;
         }
 
         private String verifyFile(FileUpload upload, int expectedSize, Function<Integer, Byte> expectedByte) {
@@ -270,6 +288,16 @@ public class MultiByteFileTest {
         @POST
         @Consumes(MediaType.MULTIPART_FORM_DATA)
         String postEmpty(@MultipartForm ClientForm form);
+
+        @Path("/type")
+        @POST
+        @Consumes(MediaType.MULTIPART_FORM_DATA)
+        String postMultipartToType(@MultipartForm ClientForm clientForm);
+
+        @Path("/type")
+        @POST
+        @Consumes("multipart/mixed")
+        String postMultipartMixedToType(@MultipartForm ClientForm clientForm);
     }
 
     public static class ClientForm {

--- a/independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java
+++ b/independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java
@@ -39,6 +39,7 @@ import org.jboss.resteasy.reactive.client.spi.MultipartResponseData;
 import org.jboss.resteasy.reactive.common.core.Serialisers;
 import org.jboss.resteasy.reactive.common.util.MultivaluedTreeMap;
 
+import io.netty.handler.codec.http.HttpHeaderValues;
 import io.netty.handler.codec.http.LastHttpContent;
 import io.netty.handler.codec.http.multipart.InterfaceHttpData;
 import io.smallrye.mutiny.Multi;
@@ -482,6 +483,19 @@ public class ClientSendRequestHandler implements ClientRestHandler {
         MultiMap multipartHeaders = multipartFormUpload.headers();
         for (String multipartHeader : multipartHeaders.names()) {
             headerMap.put(multipartHeader, multipartHeaders.getAll(multipartHeader));
+        }
+
+        if (state.getEntity().getVariant() != null) {
+            Variant v = state.getEntity().getVariant();
+            String variantContentType = v.getMediaType().toString();
+            String multipartContentType = headerMap.getFirst(HttpHeaders.CONTENT_TYPE);
+            if (multipartContentType.startsWith(HttpHeaderValues.MULTIPART_FORM_DATA.toString())
+                    && !variantContentType.startsWith(HttpHeaderValues.MULTIPART_FORM_DATA.toString())) {
+                // this is a total hack whose purpose is to allow the @Consumes annotation to override the media type
+                int semicolonIndex = multipartContentType.indexOf(';');
+                headerMap.put(HttpHeaders.CONTENT_TYPE,
+                        List.of(variantContentType + multipartContentType.substring(semicolonIndex)));
+            }
         }
 
         setVertxHeaders(httpClientRequest, headerMap);

